src/MySqlConnector/Serialization/SerializationUtility.cs

```diff
@@ -47,5 +51,30 @@ public static int GetBytesPerCharacter(CharacterSet characterSet)
 				return 1;
 			}
 		}
+
+		public static void SetKeepalive(Socket socket, uint keepAliveTimeSeconds)
+		{
+			// Always use the OS Default Keepalive settings
+			socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
```

Comment:
Following up on question above, should this option only be set if user has set keepAlive != 0?

If you think that the benefits of turning it on by default are worthwhile, then I have no objection. (I'm in favour of changing to use sensible defaults.)